Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -2017,30 +2018,40 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
             this._layoutBadgeElement = null;
         }
 
-        if (!this.representedObject.layoutContextType)
-            return;
+        if (this._scrollBadgeElement) {
+            this._scrollBadgeElement.remove();
+            this._scrollBadgeElement = null;
+        }
 
-        this._layoutBadgeElement = this.title.appendChild(document.createElement("span"));
-        this._layoutBadgeElement.className = "layout-badge";
+        if (this.representedObject.layoutFlags.includes(WI.DOMNode.LayoutFlag.Scrollable)) {
```

Comment:
We should create a `WI.settings.experimentalEnableElementBadgeScrollable` while we figure out how to handle the (likely inevitable) explosion of badges.  I'd suggest following an existing experimental setting (e.g. `experimentalEnableNetworkEmulatedCondition`) to see where we create it, add it to the Settings Tab, and how to use it.

I'm also a bit concerned that this badge could confuse developers since it's not interactable, as other badges (both existing and planned in the future) all have some sort of action when clicked.

Reply:
alternatively we could just wait for #3251 :)